### PR TITLE
When creating an SdkException without a message, inherit the "cause" exception message.

### PR DESCRIPTION
**Pull request by millems**

### CODE REVIEW

File: core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @Test
+    public void sdkExceptionMessagesDefaultToChildMessages() {
+        Exception barCause = new Exception("bar");
+        Exception nullCause = new Exception();
+
+        assertThat(SdkException.builder().build().getMessage()).isEqualTo(null);
+        assertThat(SdkException.builder().cause(nullCause).build().getMessage()).isEqualTo(null);
+        assertThat(SdkException.builder().message("foo").build().getMessage()).isEqualTo("foo");
+        assertThat(SdkException.builder().message("foo").cause(barCause).build().getMessage()).isEqualTo("foo");
+        assertThat(SdkException.builder().cause(barCause).build().getMessage()).isEqualTo("bar");
```
Comment (bmaizels) [inline core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java]: Ideally these would be separate tests so as to follow the 'unit test should test just one behavior' pattern/constraint.

Comment (millems) [reply core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java]: Improved

**Comment by spfink** [comment]
Are you submitting this under the Apache 2 License?

File: core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package software.amazon.awssdk.core.exception;
```
Comment (zoewangg) [inline core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java]: copyright header?

Comment (spfink) [reply core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java]: Clearly skipping checkstyle. If only we had a checklist for this...

Comment (millems) [reply core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionTest.java]: Fixed.

**Pull request merged**